_posts/2025-05-16-Introduction_to_bidirectional-languages-in-your-website.md

```diff
@@ -0,0 +1,127 @@
+
+There are out-of-the-box solutions that deals with margins and padding etc. automatically such as [rtlcss](https://rtlcss.com/learn/) and [tailwindcss-rtl](https://www.npmjs.com/package/tailwindcss-rtl). For demonstration purposes I will show you how to deal with them manually.
+
+When creating a website with support for both directionalities you should avoid using `margin` and `padding`. Instead we should use `margin-inline-start`, `margin-inline-end`,
+`padding-inline-start` and `padding-inline-start` which aligns items depending on the directionality.
+
+[Tailwind](https://tailwindcss.com/blog/tailwindcss-v3) offers RTL and LTR modifiers which can be used.
+
+```html
+<header className="... ltr:pl-4 rtl:pr-4"></header>
+```
+
+![[simple website 4.png]](/img/2025-05-16-introduction-to-bidirectional-languages-in-your-website/4.png)
+
+Now the margins are fixed on the header. There is still one small quirk when setting directionality for the page.
+
+I didn't translate this text just to demonstrate this special case when having LTR text in a RTL directional page.
+
+![[simple website 5.png]](/img/2025-05-16-introduction-to-bidirectional-languages-in-your-website/5.png)
+
+### Neutral characters
+
+Some Unicode characters are not associated with some directionality e.g. numbers and punctuation. These characters are considered to be weak or neutral. The [Unicode bidirectional algorithm](https://www.w3.org/International/articles/inline-bidi-markup/uba-basics) usually handles these situations automatically by looking the characters around the neutral or weak character but not always.
+
+In this case the text is in English but the language is set to Farsi and the base direction is RTL. This is why the exclamation mark is on the left side.
+
+We can fix this by using [bidirectional isolate element](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/bdi) `<bdi>` instead of e.g. `p` but as for now this isn't supported on Safari browsers. The other way is to set the directionality on the element that contains LTR text.
+
+```html
+<p dir="ltr" className="self-center text-sm text-primary">{t("ltr_text")}</p>
+```
+
+![[simple website 6.png]](/img/2025-05-16-introduction-to-bidirectional-languages-in-your-website/6.png)
+
+Now the exclamation mark is in the correct place and the website look already much better and supports RTL layouts correctly.
+
+![[simple website 7.png]](/img/2025-05-16-introduction-to-bidirectional-languages-in-your-website/7.png)
+
+## What else?
+
+There is a lot more to be considered when supporting RTL layouts with localization such as inputs, dates, numerals. In the theme of introduction I wanted to keep it simple and focus on the layout itself and some edge cases like the LTR text in RTL directional page.
+
+If there is a need for a deep dive on bidirectional text and localization I will make a another post with more info on how to tackle these cases.
```

Comment:
As this is public blog post I would maybe reword this somehow. "Perhaps these will be a topic of another blog post in the future" or something like that.